gradoop-examples/src/main/java/org/gradoop/benchmark/sampling/SamplingBenchmark.java

```diff
@@ -0,0 +1,421 @@
+
+    writeBenchmark(env, algorithm);
+
+    if (COMPUTE_STATISTICS) {
+      // compute input graph statistics
+      StatisticsRunner.main(
+        new String[]{
+            OUTPUT_PATH + OUTPUT_PATH_GRAPH_SAMPLE_SUFFIX,
+            "csv",
+            OUTPUT_PATH + GRAPH_STATISTICS_ORIGIN
+        });
+      // compute sample graph statistics
+      StatisticsRunner.main(
+        new String[]{INPUT_PATH, "csv", OUTPUT_PATH + GRAPH_STATISTICS_SAMPLE});
+    }
+  }
+
+  /**
+   * Reads the given arguments from command line
+   *
+   * @param cmd command line
+   */
+  private static void readCMDArguments(CommandLine cmd) {
+    INPUT_PATH          = cmd.getOptionValue(OPTION_INPUT_PATH);
+    SELECTED_ALGORITHM  = Integer.parseInt(cmd.getOptionValue(OPTION_SELECTED_ALGORITHM));
+    CONSTRUCTOR_PARAMS  = cmd.getOptionValues(OPTION_CONSTRUCTOR_PARAMS);
+    OUTPUT_PATH         = cmd.getOptionValue(OPTION_OUTPUT_PATH, OUTPUT_PATH_DEFAULT);
+    COMPUTE_STATISTICS  = cmd.hasOption(OPTION_COMPUTE_STATISTICS);
+  }
+
+  /**
+   * Method to crate and add lines to a benchmark file
+   * @param env given ExecutionEnvironment
+   * @param sampling sampling algorithm under test
+   * @throws IOException exception during file writing
+   */
+  private static void writeBenchmark(ExecutionEnvironment env, SamplingAlgorithm sampling)
+      throws IOException {
+    String head = String.format("%s|%s|%s|%s|%s%n",
+      "Parallelism", "Dataset", "Algorithm", "Params", "Runtime [s]");
+
+    // build log
+    String samplingName = sampling.getClass().getSimpleName();
+    String tail = String.format("%s|%s|%s|%s|%s%n",
+      env.getParallelism(),
+      INPUT_PATH.substring(INPUT_PATH.lastIndexOf(File.separator) + 1),
+      samplingName,
+      String.join(", ", CONSTRUCTOR_PARAMS),
+      env.getLastJobExecutionResult().getNetRuntime(TimeUnit.SECONDS));
+
+    File f = new File(OUTPUT_PATH + OUTPUT_PATH_BENCHMARK_SUFFIX);
+    if (f.exists() && !f.isDirectory()) {
+      FileUtils.writeStringToFile(f, tail, true);
+    } else {
+      PrintWriter writer = new PrintWriter(OUTPUT_PATH + OUTPUT_PATH_BENCHMARK_SUFFIX, "UTF-8");
+      writer.print(head);
+      writer.print(tail);
+      writer.close();
+    }
+  }
+
+  /**
+   * Tries to instantiate a SamplingAlgorithm defined by the enum Algorithm. Maps the
+   * value of SELECTED_ALGORITHM and the length of CONSTRUCTOR_PARAMS to a specific constructor.
+   * Constructors that take non-primitive data types as input are not provided.
+   * If the value of SELECTED_ALGORITHM does not match any ordinal of the enum, or the length of
+   * CONSTRUCTOR_PARAMS does not match any signature of a specified SamplingAlgorithm, an exception
+   * is thrown.
+   *
+   * @return SamplingAlgorithm specified by const SELECTED_ALGORITHM
+   */
+  private static SamplingAlgorithm instantiateSelectedSamplingAlgorithm() {
+
+    if (SELECTED_ALGORITHM < 0 || SELECTED_ALGORITHM > Algorithm.values().length - 1) {
```

Comment:
In this Class you should provide a `perfomSanityCheck` function. Which can check the given parameters. 

e.g. `GroupingBenchmark` part of the sanity check

```
  /**
   * Checks if the minimum of arguments is provided
   *
   * @param cmd command line
   */
  private static void performSanityCheck(final CommandLine cmd) {
    if (!cmd.hasOption(OPTION_INPUT_PATH)) {
      throw new IllegalArgumentException("Define a graph input directory.");
    }
  }
```

You could add a sanity check within the SamplingBuilder as well.